wgsl/index.bs

```diff
@@ -671,7 +671,179 @@ Any other context.  Evaluates to *P.Read()*, yielding a value of *P*’s pointee
 type.
 </table>
 
-## Texture and Sampler Types TODO ## {#texture-types}
+## Texture and Sampler Types ## {#texture-types}
+
+A <dfn noexport>texel</dfn> is a scalar or vector used as the smallest independently accessible element of a [=texture=].
+The word *texel* is short for *texture element*.
+
+A <dfn noexport>texture</dfn> is a collection of texels supporting special operations useful for rendering.
+In [SHORTNAME], those operations are invoked via texture builtin functions.
+See [[#texture-builtin-functions]] for a complete list.
+
+A [SHORTNAME] texture corresponds to a [[WebGPU#gputexture|WebGPU GPUTexture]].
+
+A texture is either arrayed, or non-arrayed:
+
+* A <dfn noexport>non-arrayed texture</dfn> is a grid of texels. Each texel has a unique grid coordinate.
+* An <dfn noexport>arrayed texture</dfn> is a homegeneous array of grids of texels.
+    In an arrayed texture, each texel is identified with its unique combination of array index and grid coordinate.
+
+A texture has the following features:
+
+: texel format
+:: The data in each texel. See [[#texel-formats]]
+:  dimensionality
+:: The number of dimensions in the grid coordinates, and how the coordinates are interpreted.
+    The number of dimensions is 1, 2, or 3.
+    In some cases the third coordinate is decomposed so as to specify a cube face and a layer index.
+: size
+:: The extent of grid coordinates along each dimension
+: mipmap levels
+:: The mipmap level count is at least 1 for sampled textures, and equal to 1 for storage textures.<br>
+    Mip level 0 contains a full size version of the texture.
+    Each successive mip level contains a filtered version of the previous mip level
+    at half the size (within rounding) of the previous mip level.<br>
+    When sampling a texture, an explicit or implicitly-computed level-of-detail is used
+    to select the mip levels from which to read texel data.  These are then combined via
+    filtering to produce the sampled value.
+: arrayed
+:: whether the texture is arrayed
+: <dfn noexport>array size</dfn>
+:: the number of homogeneous grids, if the texture is arrayed
+
+A texture's representation is typically optimized for rendering operations.
+To achieve this, many details are hidden from the programmer, including data layouts, data types, and
+internal operations that cannot be expressed directly in the shader language.
+
+As a consequence, a shader does not have direct access to the texel storage within a texture variable.
+Instead, use texture builtin functions as follows:
+
+* Within the shader:
+    * Declare a module-scope variable in the `uniform_constant` storage class,
```

Comment:
Storage class name needs an update.
In WebGPU spec we'd reference those as something like {{GPUStorageClass/"handle'"}}, which allows `bikeshed` to generate a reference link

Reply:
I'll fix this to 'handle' now.

The  {{Storage classes/handle}} construct requires a bigger change to the storage class table.

It looks like the syntax that will work here is    [=storage classes/handle=].
I'd like to do the whole storage classes table at once, but that will generate a conflict with this PR.  So I'll do that in a followup.

Reply:
I filed #1232 for followup